packages/utils/src/lib/reports/utils.ts

```diff
@@ -251,10 +247,10 @@ export async function loadReport<T extends Format>(
 
   if (format === 'json') {
     const content = await readJsonFile(filePath);
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-return
     return reportSchema.parse(content) as any;
   }
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-return
   return readTextFile(filePath) as any;
```

Comment:
This looks nasty :grimacing: Why are the `as any`s needed?

Reply:
Apparently, TypeScript does not understand how we distinguish the type and therefore screams.
![image](https://github.com/code-pushup/cli/assets/6306671/50bc2fd5-befb-4fa7-86a4-80f59679db0f)

Reply:
I see. Yeah, this is pretty common when the return type depends on a generic, some casting is needed.

But if you cast as the return type instead, you can get rid of the ESLint disable comments.

```ts
  if (format === 'json') {
    const content = await readJsonFile(filePath);
    return reportSchema.parse(content) as LoadedReportFormat<T>;
  }
  const text = await readTextFile(filePath);
  return text as LoadedReportFormat<T>;
```